drafts/arcalinea-tx-expiry/draft.rst

```diff
@@ -0,0 +1,83 @@
+::
+
+  ZIP: 203
+  Title: Transaction Expiry
+  Author: Jay Graber <jay@z.cash>
+  Category: Consensus
+  Created: 2018-01-09
+  License: MIT
+
+
+Abstract
+===========
+
+This is an Standards ZIP describing a new consensus rule to set an expiration time for transactions that are not mined to be removed from the mempool.
+
+Motivation
+===========
+
+Transactions that have insufficient fees are often not mined. This indeterminism is a source of confusion for users and wallets. Allowing a transaction to set a block height after which it cannot be mined would provide certainty around how long a transaction has to confirm before it is rejected by the network and must be re-sent.
+
+Advantages include optimizing mempool performance by removing transactions that will not be mined, and potentially simplifying bidirectional payment channels by reducing the need to store and compress revocations for past states, since transactions not committed to the chain could expire and become invalid after a period of time.
+
+If the expiry is at block height N, then the transaction must be included in block N or earlier. Block N+1 will be too late, and the transaction will be removed from the mempool.
+
+The new consensus rule will enforce that the transaction will not be considered valid if included in block of height greater than N, and blocks that include transactions which have expired will not be considered valid.
```

Comment:
s/transactions which have expired/expired transactions/

(See https://en.oxforddictionaries.com/usage/that-or-which . It would be correct to say "blocks that include transactions that have expired", but it sounds better to avoid the double "that".)